### PR TITLE
make MySQL56-flavor schema queries forward-compatible

### LINKED ISSUE #11312: -db_flavor FilePos doesn't work well out-of-the-box with external MySQL 8.0 tablets
@aquarapid reported this to me.

# Description

Starting a tablet on top of:

 * An external MySQL 8.0 data source
 * With `-db_flavor=FilePos`

The tablet does not reach ready state, and logs messages similar to these:

```
E0921 05:57:46.351677       1 tm_state.go:331] Cannot start query service: Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'information_schema.tables.TABLE_TYPE' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by (errno 1055) (sqlstate 42000) during query: SELECT table_name, table_type, unix_timestamp(create_time), table_comment, SUM( data_length + index_length), SUM( data_length + index_length)
```

Note that the tablet is trying to execute MySQL queries that the external MySQL doesn't support.

# Details

The issue seems to be that the `filePosFlavor` is [hard-coded](https://github.com/vitessio/vitess/blob/0b6c64442/go/mysql/flavor_filepos.go#L329-L332) to use a MySQL 5.6 flavor information schema query:

```
// baseShowTablesWithSizes is part of the Flavor interface.
func (*filePosFlavor) baseShowTablesWithSizes() string {
	return TablesWithSize56
}
```

Removing `only_full_group_by` from the SQL mode fixes the issue and allows the tablet to come online, but we think it would be better if we didn't have to do this.

### PR DESCRIPTION
**Pull request by maxenglander**
## Description

The various SQL flavors (56, 57, 80, MariaDB, FilePos) use different queries for schema inspection. The FilePos flavor is using a 56 flavor query which isn't forward compatible with MySQL 80.
## Related Issue(s)

This commit updates the 56 flavor to be forward-compatible with 57, 80. s/o @mattlord for the suggested fix.

Fixes #11312.

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by mattlord — APPROVED** [review]
Thanks! ❤️ I'll let Query Serving review too and they can merge.

**Review by dbussink — APPROVED** [review]
Thnx for the fix!

**Review by deepthi — APPROVED** [review]
Technically query-serving but we don't need to wait for their review.
The approvals we have so far are sufficient.

**Pull request merged**
